libs/base/ui/structure/text/text.component.ts

```diff
@@ -11,6 +11,76 @@ export class TextComponent extends LitElement implements TextProperties {
 
   protected render(): TemplateResult | void {
     if (!this.content) return;
-    return html`${unsafeHTML(this.content)}`;
+    const c = this.appendStyleToHeaders(this.content);
+    return html`${unsafeHTML(c)}`;
+  }
+
+  /**
+   * Modifies an HTML string by adding custom inline styles to specific HTML elements
+   * based on their tags, classes, and screen sizes. The styles align roughly with the
+   * Oryx design system without reusing the design system components directly.
+   *
+   * @param htmlString - The HTML string to be modified.
+   * @returns The modified HTML string with custom inline styles.
+   */
+  protected appendStyleToHeaders(htmlString: string): string {
+    const headerRegex = /<(\w+)([^>]*)>(.*?)<\/\1>|<(\w+)([^>]*)\/>/gis;
+
+    const modifiedHtmlString = htmlString.replace(
+      headerRegex,
+      (match, tag, attributes, content) => {
+        if (!tag) return match;
+        const classes = attributes?.match(/class="(.*?)"/i)?.[1].trim();
```

Comment:
I would extract all these regexps to the separate consts and put the at the top. At least this huge amount of processing become a bit more readable

Reply:
And should all these regexps be case-insenitive?

Reply:
I'd say yes, they need to be case insensitive, as we do not know if third party CMSes or other (manua) content providers are strict on case.

Reply:
Good suggestion with externalizing the regex.

Reply:
refactored heavily so closing this.